WordPress/gutenberg
# Create Block: Adds `--target-dir` flag to allow the tool to target where to scaffold

## What?
This PR adds the `--target-dir` flag to allow users to target where the tool should scaffold either the entire plugin or the block files. This flag is not used when the tool is in `--no-plugin` mode

Closes #44717

## Why?
Being able to target where to scaffold the files is something that has been missing from the tool. By adding this flag extenders would be able to incorporate create-block into their scaffolds/workflow much easer.

For example, many projects use `wp-content` as the root directory and keep a package.json there. Currently, anyone wanting to use the create-block tool, would need to move the `plugins` directory and run it there. With this change, a script could be created to run the tool in place.

### Script
```
"scaffold-block": "npx @wordpress/create-block --target-dir plugins"
```

### Invoke
```
 npm run scaffold-block -- block-name
```
Another change here is now  when running the tool in `--no-plugin` mode, the resulting files are scaffolded into a sub-directory of the `folderName` property ( `./src` by default ) so now when adding a new block to an existing plugin the user no longer needs to move the the `folderName` directory. 

## How?
I am introducing a `targetDir `variable with the default value of `./` and prepending all of the paths with its value. 

## Testing Instructions
1. Checkout this branch
2. `cd` into `packages/create-block'
3. Run `node index.js test-no-target`
4. Confirm that a plugin is created at the root with the block files in a `src` directory
5. Run `node index.js test-with-target --target-dir target-directory`
6. Confirm that a plugin is created the `target-directory` folder with the block files in a `src` directory
